dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/JdbcTrackedEntityChangeLogStore.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2004-2024, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.tracker.export.trackedentity;
+
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.hisp.dhis.program.UserInfoSnapshot;
+import org.hisp.dhis.tracker.export.trackedentity.TrackedEntityChangeLog.Change;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+import org.springframework.stereotype.Repository;
+
+@Repository("org.hisp.dhis.tracker.export.trackedentity.JdbcTrackedEntityChangeLogStore")
+@RequiredArgsConstructor
+public class JdbcTrackedEntityChangeLogStore {
+
+  private final NamedParameterJdbcTemplate namedParameterJdbcTemplate;
+
+  private static final RowMapper<TrackedEntityChangeLog> customTrackedEntityChangeLogRowMapper =
```

Comment:
Is there a reason why we keep using Spring jdbc? I think if there isn't a specific reason we should use the JPA standards with named queries which gives the same support plus it is a standard previous to using Spring in general i think

Reply:
This was a personal choice of mine; I find it easier to work with JDBC rather than the entity manager or native queries. From my experience, I had some trouble, particularly when dealing with Collections. 
Before I started implementing this, I took a look at the tracker code base and noticed that we are using JDBC, HQL, and criteria queries in our endpoints. I don't think we have agreed on using one or the other; perhaps we could discuss this further and find common ground.

Reply:
> Is there a reason why we keep using Spring jdbc?
It spread from the EventStore which existed before @muilpp and I joined. I as well prefer full control over the queries so writing SQL is what I would go for. Feel free to present the differences and the advantages of what you are suggesting.

We do not have time to discuss and adapt this right now though 😬

Reply:
I didn't mean to adapt it before the soft freeze, but after the release it would be good to discuss this because it keeps coming back. To at least we could be in sync in our team.